pkg/namespacestore/validation.go

```diff
@@ -28,6 +28,11 @@ func ValidateNamespaceStore(nsStore *nbv1.NamespaceStore) error {
 		return ValidateNsStoreNSFS(nsStore)
 
 	case nbv1.NSStoreTypeAWSS3:
+		if nsStore.Spec.AWSS3.AWSSTSRoleARN != nil {
```

Comment:
you also need to handle a `nil` secret in `ValidateNSEmptySecretName` which is called above

Reply:
Done